### DISCUSSION TITLE
220523 오찌 인터뷰 피드백

### DISCUSSION THREAD

**Discussion by YJGwon** [woowacourse-study/2022-lv2-effective-java-interview#12] [General]
### 인터뷰어 -> 인터뷰이

> 점수를 매겨주셔도 좋고, 체크 형식으로 작성해주신 뒤 짧은 코멘트를 남겨주셔도 좋습니다. 해당 항목에 대해 자유롭게 의견을 남겨주세요.

- 말하기 태도, 어조가 적절했나요?
- 전달하고자 하는 바를 명료하게 전달했나요?
- 본인만의 주관이 잘 드러났나요?
- 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- 모르는 부분에 대해 잘 대처했나요?

### 인터뷰이 -> 인터뷰어

> 인터뷰어로써의 역량을 키우려는 것이 아닌 저희 스터디의 원활한 진행을 위한 피드백입니다. 각 크루가 아닌 전반적인 진행 상황에 대해 의견을 남겨주세요!

- 사전에 공유한 내용이 잘 전달되었나요?
- 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- 새로운 관점을 배울 수 있었나요?

---

### 질문 리스트
- Lombok의 Builder Annotation은 필수 field와 선택 field를 설정해줄 수 있는것인지?
- 애초에 생성자의 매개변수(field)가 왜 많아졌는가, 줄일 수 없는가에 대한 고민이 선행되어야 하는것은 아닐지?
- new 키워드 사용을 없애고 정적 팩토리 메서드를 만들어주었는데, 가독성 이외에 어떤 장점이 있을지?
- 객체 지향 개발에서 production에 드는 비용이 어떻게 고려되어야 한다고 생각하는지?
    - 초기 production 비용을 줄이기 위한 고민이 어느정도 비중으로 이루어져야 한다고 생각하는지?
- 이번 미션은 만들어둔 코드를 바탕으로 요구사항이 계속 추가되는 미션이었음에도 기한때문에 정상 동작하는 코드를 빠르게 만들어내는 것을 선택했다. 그 후 느낀 점이 있었는지, 그렇다면 현업에서 비슷한 상황이 있을때는 어떻게 대처할 것인지?
- Builder 패턴을 써야겠다라는 생각이 드는 기준?
- 빌더 패턴을 유지하면서도 필수 필드에 대한 제약을 걸어줄 방법에대해 고민해보았는지?
- git reset 이후 개발 비용이 드는 문제로 builder를 포기했는데, Lombok을 사용하지 않은 이유는?
    - 미션에 외부 라이브러리 사용에 대한 제약이 없어도 쓰지 않을것인지?
- git reset이 된 이후에도 IntelliJ 파일 수정 기록에는 남아있었을텐데, 활용해보셨는지?
    - 파일 우클릭 - local history

**Comment by tco0427** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
- [x] 본인만의 주관이 잘 드러났나요?
  -> 경험 기반으로 본인의 생각을 잘 전달해주셨습니다.
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
  -> 넵! 책의 내용을 그대로 수용하는게 아니라, 정적 팩토리를 함께 도입하는 등 본인만의 경험을 살린 것 같습니다.
- [x] 모르는 부분에 대해 잘 대처했나요?
  -> 없던데요...

**Comment by kyukong** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
- [x] 본인만의 주관이 잘 드러났나요?
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- [x] 모르는 부분에 대해 잘 대처했나요?

모르는 부분은 없었지만 모든 질문에 잘 대처하여 체크했습니다.ㅎㅎ 말의 속도가 빠른 편인데 발음도 정확하고 기승전결을 제대로 지켜서 말하기 때문에 듣기 좋았습니다!

**Comment by yeon-06** [comment]
자신있게 자신의 생각을 말씀해주시는게 좋았습니다. 발표 자료도 이론적인 개념을 정리하기 보다는 오찌의 생각을 미리 볼 수 있었던 자료라서 재미있게 읽었습니다. 질문을 몇 개 적어두었는데 하나를 물어보면 두세개의 의문이 해결되어서 물어볼게 별로 없었네요😅

**Comment by YJGwon** [comment]
- [x] 말하기 태도, 어조가 적절했나요?
    - 발음이 약간 불문명한 경향이 있으나 attitude가 좋아서 괜찮게 느껴지는 듯 합니다
- [x] 전달하고자 하는 바를 명료하게 전달했나요?
- [x] 본인만의 주관이 잘 드러났나요?
    - 단순히 요구사항에 맞춘 구현만 하는 것이 아니라 이 것을 구현하는 **과정**에서 본인이 얻고싶은 **목표**에 대해 중심을 가지고 있는 것이 멋지네요
- [x] 책의 내용을 단순히 수용하지 않고 본인의 경험과 잘 연결지어 이해했나요?
- [x] 모르는 부분에 대해 잘 대처했나요?

**Comment by ulimy** [comment]
해당 부분에 대해 경험해보았음이 느껴져 좋았습니다. 적절한 손동작이 신뢰감을 높여준 것 같아요. 다만 전체적으로 이.. 그.. 저.. 등이나 이렇게 저렇게 등의 표현을 많이 사용하시는데 사용하지 않도록 주의하시면 이야기가 더 분명하게 들릴 것 같습니다!

**Comment by Ohzzi** [comment]
- [x] 사전에 공유한 내용이 잘 전달되었나요?
- [x] 질문의 방향이 단순히 fact check가 아닌 의견을 끌어내는 방향이었나요?
- [x] 생각해보지 못했던 부분에 대해 생각해볼 수 있었나요?
- [x] 새로운 관점을 배울 수 있었나요?

파일 우클릭 - local history 너무 충격적이게 좋았습니다!
가장 좋았던 건 다양한 관점에서 질문을 주셨던 부분인 것 같아요. 설계에 대한 의문부터, 객체지향 방법론 등의 생각해보지 못했던 관점에 대한 질문들이 들어와서 많은 것을 생각해볼 수 있는 시간이었습니다!
